### PR TITLE
Check if the shop logo exists

### PR DESCRIPTION
**Pull request by YordanSoares**
Closes: #421

### CODE REVIEW

File: includes/documents/abstract-wcpdf-order-document.php [hunk 1]
```diff
@@ -640,8 +640,10 @@ public function header_logo() {
 
 				if ( apply_filters( 'wpo_wcpdf_use_path', true ) && file_exists( $attachment_path ) ) {
 					$src = $attachment_path;
-				} else {
+				} elseif ( file_exists( $attachment_src ) ) {
```
Comment (alexmigf) [inline includes/documents/abstract-wcpdf-order-document.php:643]: The `$attachment_src` is an URL not a local path, I think you must use this function: [wp_remote_head](https://developer.wordpress.org/reference/functions/wp_remote_head/)

The you should check if the response header code is `200`, which corresponds to success (exists).

**Review by alexmigf — CHANGES_REQUESTED** [review]

File: includes/documents/abstract-wcpdf-order-document.php [hunk 2]
```diff
@@ -624,27 +624,31 @@ public function get_header_logo_height() {
...
-				$attachment_src = $attachment[0];
-				$attachment_width = $attachment[1];
+				$attachment_src    = $attachment[0];
+				$attachment_width  = $attachment[1];
 				$attachment_height = $attachment[2];
 
 				if ( apply_filters( 'wpo_wcpdf_use_path', true ) && file_exists( $attachment_path ) ) {
 					$src = $attachment_path;
-				} else {
+				} elseif ( wp_remote_head( $attachment_src, [ 'sslverify' => false ] )['response']['code'] === 200 ) {
```
Comment (alexmigf) [inline includes/documents/abstract-wcpdf-order-document.php:645]: I think you probably should check the response first before asking for the keys, like this:

```php
elseif( ( is_array( $head = wp_remote_head( $attachment_src, [ 'sslverify' => false ] ) ) ) && $head['response']['code'] == 200 ) )
```

Comment (YordanSoares) [reply includes/documents/abstract-wcpdf-order-document.php]: Done here: 0601f05

**Comment by alexmigf** [comment]
@YordanSoares I have made some adjustments, let me know what you think.

**Comment by YordanSoares** [comment]
@alexmigf your improvements are great: Now it checks almost all what comes to my mind!

Also, I ran some tests and haven't any issues 👍

**Pull request merged**